frame/bounties/src/lib.rs

```diff
@@ -208,6 +208,165 @@ decl_storage! {
+		fn random<R: Rng>(rng: &mut R) -> Self;
+	}
+
+	// Implementations of primitives will be added by (a nicer) macro wherever the Randomize trait
+	// is actually defined. Implementations for types like `BoundedVec` and `StorageMap` will be
+	// added in the crates in which they are defined.
+
+	macro_rules! impl_randomize_for_primitive {
+		($t:ty) => {
+			impl Randomize for $t {
+																			fn random<R: Rng>(rng: &mut R) -> Self {
+																				rng.gen()
+																			}
+																		}
+		};
+	}
+
+	impl_randomize_for_primitive!(u8);
+	impl_randomize_for_primitive!(u16);
+	impl_randomize_for_primitive!(u32);
+	impl_randomize_for_primitive!(u64);
+	impl_randomize_for_primitive!(u128);
+	impl_randomize_for_primitive!(i8);
+	impl_randomize_for_primitive!(i16);
+	impl_randomize_for_primitive!(i32);
+	impl_randomize_for_primitive!(i64);
+	impl_randomize_for_primitive!(i128);
+
+	// We'll need to provide a derive macro for user-defined types like these.
+
+	impl<AccountId, BlockNumber> Randomize for BountyStatus<AccountId, BlockNumber>
+	where
+		AccountId: Randomize,
+		BlockNumber: Randomize,
+	{
+		fn random<R: Rng>(rng: &mut R) -> Self {
+			// Obviously a real implementation would pick a random variant, but this is just for
+			// demo purposes.
+			// https://xkcd.com/221/
+			Self::Active { curator: AccountId::random(rng), update_due: BlockNumber::random(rng) }
+		}
+	}
+
+	impl<AccountId, Balance, BlockNumber> Randomize for Bounty<AccountId, Balance, BlockNumber>
+	where
+		AccountId: Randomize,
+		Balance: Randomize,
+		BlockNumber: Randomize,
+	{
+		fn random<R: Rng>(rng: &mut R) -> Self {
+			Self {
+				proposer: AccountId::random(rng),
+				value: Balance::random(rng),
+				fee: Balance::random(rng),
+				curator_deposit: Balance::random(rng),
+				bond: Balance::random(rng),
+				status: BountyStatus::random(rng),
+			}
+		}
+	}
+
+	// We _won't_ implement this for a real vector, but it makes life simpler for this
+	// demonstration. LIMIT here stands in for `BoundedVec::LIMIT`.
+	const LIMIT: usize = 20;
+	impl<T> Randomize for Vec<T>
+	where
+		T: Randomize,
+	{
+		fn random<R: Rng>(rng: &mut R) -> Self {
+			let mut v = Vec::with_capacity(LIMIT);
+			for _ in 0..LIMIT {
+				v.push(T::random(rng));
+			}
+			v
+		}
+	}
+
+	// The StorageMap implementation always inserts 300,000 items, on the basis that this should be
+	// a reasonable quantity for miscellaneous "Items" which would get put in a storagemap in a real
+	// blockchain. This could use some more thought, and maybe a different api.
+	//
+	// Still, the critical element is that generating all this data happens _once_, not once per
+	// iteration of the benchmarker. I'll have to check to be sure, but I believe that the
+	// benchmarker already takes advantage of the layered nature of storage, and just clears out all
+	// edits made per iteration. As long as this data is committed before the bench run starts, it
+	// will be preserved.
```

Comment:
This is true for benchmark repeats, but not for benchmark steps or for multiple extrinsics.

I think if you take this approach, we will need to modify the benchmarking pipeline. 

I also don't think this is the right approach in the long term. As mentioned in chat, I think a better option is to track the number of "maps" we put in storage, and give each one some pre-determined POV weight.